### ISSUE TITLE
enhancing the logging

### ISSUE THREAD

**Issue by TobiasHH** [pushpak1300/cypress-mailpit#50]
Actual the usage of the mailpit logs directly the internal commands of the custom commands.

Cypress gives the possibility to use custom logging for custom commands so that the log gets more readable.

**Comment by TobiasHH** [comment]
please tell me when to implement

**Comment by pushpak1300** [comment]
Yes. @TobiasHH before implementing let's think a unify way to do logging in all the commands with minimal breaking change.

WhatI'm thinking there should be env variable by which we should control if we should log or not for all command instead of indiviual option.

The second thing is if we enable logging then what all things do you think we should log ?
The request,  the response what exactly ?

**Comment by TobiasHH** [comment]
Good question.

https://docs.cypress.io/api/cypress-api/cypress-log

```
Cypress.Commands.add('setSessionStorage', (key, value) => {
  // Turn off logging of the cy.window() to command log
  cy.window({ log: false }).then((window) => {
    window.sessionStorage.setItem(key, value)
  })

  const log = Cypress.log({
    name: 'setSessionStorage',
    // shorter name for the Command Log
    displayName: 'setSS',
    message: `${key}, ${value}`,
    consoleProps: () => {
      // return an object which will
      // print to dev tools console on click
      return {
        Key: key,
        Value: value,
        'Session Storage': window.sessionStorage,
      }
    },
  })
})
```

"WhatI'm thinking there should be env variable by which we should control if we should log or not for all command instead of indiviual option."

I don't think you need an env variable? The last time I added the parameter only to hide the internal called command.

```
	mailpitGetEmailsByTo(email: string, start = 0, limit = 50): Cypress.Chainable<MessagesSummary> {
		return this.mailpitSearchEmails(`to:${email}`, start, limit);
	}
```

this is displaying mailpitGetEmailsByTo and mailpitSearchEmails in the command log also if only mailpitGetEmailsByTo is called.

```
Cypress.Commands.add('setSessionStorage', (key, value) => {
  // Turn off logging of the cy.window() to command log
  cy.window({ log: false }).then((window) => {
    window.sessionStorage.setItem(key, value)
  })
```

Here you can see the log:false property so that the window command is not shown anymore when the setSessionStorage command is executed. The command log only displays setSessionStorage and not the internal window.

The console log I use only seldom.
I would log,
- the input value like the Address, Subject, search
- the Yielded Object like all Cypress commands do. i.e. MessagesSummary ...
- maybe the messages count

I would not log the start, limit, timeout parameter

**Comment by TobiasHH** [comment]
maybe this is an inspiration https://github.com/testing-library/cypress-testing-library/blob/main/src/index.js
